beta/nncf/tensorflow/layers/wrapper.py

```diff
@@ -118,7 +118,7 @@ def build(self, input_shape=None):
 
     def call(self, inputs, training=None):
         if training is None:
-            training = tf.keras.backend.learning_phase()
+            training = tf.cast(tf.keras.backend.learning_phase(), tf.bool)
```

Comment:
This works both in eager mode whet learning_phase is int and for graph mode when learning_phase is tf.Tensor

Reply:
Looks like casting change all the reference models.

Smart_cond can handle int values with `and` operator in both tensor and eager mode. I suggest to keep learning phase as int type.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/backend.py#L342-L351
``` 
def learning_phase():
 """Returns the learning phase flag.
  The learning phase flag is a bool tensor (0 = test, 1 = train)
  to be passed as input to any Keras function
  that uses a different behavior at train time and test time.
  Returns:
      Learning phase (scalar integer tensor or Python integer).
  """
```